device/iot-device-samples/multiplexing-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/ClientManagerBase.java

```diff
@@ -0,0 +1,220 @@
+        // If the lastKnownConnectionStatus is not in a CONNECTED state it can mean two things:
+        // 1: the status is CONNECTING, in which case there is nothing to be done at this time.
+        // 2: the status is DISCONNECTED, in which case connection cannot be re-established.
+        if (lastKnownConnectionStatus == ConnectionStatus.CONNECTED)
+        {
+            new Thread(new Runnable() {
+                @Override
+                public void run() {
+                    log.debug("Attempting reconnect for client: " + getClientId() + " ...");
+                    synchronized (lastKnownConnectionStatus)
+                    {
+                        if (lastKnownConnectionStatus == ConnectionStatus.CONNECTED)
+                        {
+                            try
+                            {
+                                closeClient();
+                            }
+                            catch (Exception ex)
+                            {
+                                log.warn("Client " + getClientId() + " closeNow failed.", ex);
+                            }
+                            finally
+                            {
+                                lastKnownConnectionStatus = ConnectionStatus.CONNECTING;
+                            }
+                        }
+                        else
+                        {
+                            log.debug("Client `" + getClientId() + "` is currently connecting; skipping...");
+                            return;
+                        }
+                    }
+
+                    // The client is now closed and the connection status is CONNECTING. Connection can be established now.
+                    try
+                    {
+                        establishConnection();
+                    }
+                    catch (IOException ex)
+                    {
+                        log.error("Exception thrown while opening client instance: " + getClientId(), ex);
+                    }
+                }
+            }).start();
+        }
+    }
+
+    public void establishConnection() throws IOException
+    {
+        // Device client does not have retry on the initial open() call. Will need to be re-opened by the calling application
+        // Lock the lastKnownConnectionStus so no other process will be able to change it while the client manager is attempting to open the connection.
+        synchronized (lastKnownConnectionStatus)
+        {
+            while (lastKnownConnectionStatus == ConnectionStatus.CONNECTING)
+            {
+                // If the client has dependencies to another client (in this case it could be the multiplexing client) we have to wait to make sure the
+                // dependent connection is established first.
+                if (dependencyConnectionStatusTracker != null && dependencyConnectionStatusTracker.getConnectionStatus() == ConnectionStatus.CONNECTING)
```

Comment:
Same idea as above, splitting conditions on multiple lines; it will also reduce line length here which is getting a bit long